File: TaskTracker.Core/src/Models/ResponseModels/WorkSpaceModel.cs
using TaskTracker.Core.src.Enums;

namespace TaskTracker.Core.src.Models.ResponseModels
{
    public class WorkSpaceModel
    {
        public int Id { get; set; }

        /// <summary>
        /// Название рабочего пространства
        /// </summary>
        public string Name { get; set; } = null!;
        public WorkSpaceType WorkSpaceType { get; set; }

        /// <summary>
        /// Ссылка на управляющего компании
        /// </summary>
        public int DirectorUserId { get; set; }

        public UserTeamRole TeamRole { get; set; }

        //Все поля ниже заполняются, если WorkSpaceType - Company

        /// <summary>
        /// Страна, в которой компания зарегистрирована
        /// </summary>
        public int? Country { get; set; }

        /// <summary>
        /// Дата регистрации в UTC
        /// </summary>
        public string? RegistrationDate { get; set; }

        /// <summary>
        /// Юр. адрес
        /// </summary>
        public string? Address { get; set; }

        public string? INN { get; set; }
    }
}

File: ui/task-tracker/src/app/shared/model/workSpaceModel.ts
import { UserTeamRole } from "../enums/user-team-role";
import { WorkspaceType } from "../enums/work-space-type";
import { WorkspaceReviewStatus } from "../enums/workspace-review-status";

export class WorkspaceModel{

    public id! : number;
    
    /**  Название рабочего пространства */
    public name! : string;
    
    public workspaceType! : WorkspaceType;
    
    /** Ссылка на управляющего компании */
    public directorUserId! : number;

    public teamRole!: UserTeamRole;
    
    //Все поля ниже заполняются, если WorkspaceType - Company
    
    /** Страна, в которой компания зарегистрирована */
    public country : number | undefined;
    
    /** Дата регистрации в UTC */
    public registrationDate : string | undefined;
    
    /** Юр. адрес */
    public address : string | undefined;
    
    public inn : string | undefined;

    public reviewStatus: WorkspaceReviewStatus | undefined;
}